skills_ml/algorithms/occupation_classifiers/train.py

```diff
@@ -0,0 +1,131 @@
+from sklearn.model_selection import GridSearchCV
+from sklearn.preprocessing import LabelEncoder
+from sklearn.model_selection import KFold, StratifiedKFold
+
+from skills_ml.storage import FSStore
+from skills_ml.algorithms.string_cleaners.nlp import NLPTransforms
+from skills_ml.ontologies.onet import build_onet, majorgroupname
+
+import numpy as np
+import importlib
+
+
+def get_all_soc():
+    onet = build_onet()
+    occupations = onet.occupations
+    soc = []
+    for occ in occupations:
+        if 'O*NET-SOC Occupation' in occ.other_attributes['categories']:
+            soc.append(occ.identifier)
+
+    return soc
+
+class OccupationClassifierTrainer(object):
+    """Trains a series of classifiers using the same training set
+    Args:
+        matrix ()
+        storage (skills_ml.storage)
+        target_variable (str)
+        embedding_model (gensim.model)
+        k_folds (int)
+        n_jobs (int)
+    """
+    def __init__(self, matrix, k_folds, grid_config=None, storage=None,
+                 random_state_for_split=None, scores=['accuracy'], n_jobs=3):
+        self.matrix = matrix
+        self.storage = FSStore() if storage is None else storage
+        self.target_variable = self.matrix['target_variable']
+        self.k_folds = k_folds
+        self.n_jobs = n_jobs
+        self.grid_config = self.default_grid_config if grid_config is None else grid_config
+        self.cls_cv_result = {}
+        self.scores = scores
+        self.best_classifiers = {}
+        self.random_state_for_split = random_state_for_split
+
+    @property
+    def default_grid_config(self):
+        return {
+                'sklearn.ensemble.ExtraTreesClassifier': {
+                    'n_estimators': [100, 500],
+                    'criterion': ['gini', 'entropy'],
+                    'max_depth': [1, 5, 10, 20, 50],
+                    'max_features': ['sqrt', 'log2'],
+                    'min_samples_split': [2, 5, 10]
+                    },
+                'sklearn.ensemble.RandomForestClassifier': {
+                    'n_estimators': [100, 500],
+                    'criterion': ['gini', 'entropy'],
+                    'max_depth': [1, 5, 10, 20, 50],
+                    'max_features': ['sqrt', 'log2'],
+                    'min_samples_split': [2, 5, 10]
+                    }
+                }
+
+    def train(self):
+        """Fit a model to a training set. Works on any modeling class that
+        is vailable in this package's environment and implements .fit
+        """
+        X = self.matrix['X']
+        y = self.matrix['y']
+        for score in self.scores:
+            for class_path, parameter_config in self.grid_config.items():
```

Comment:
This loop is the part that takes a while, so we probably want to log a bunch of things